sql/importer: TestImportWorkerFailure failed

sql/importer.TestImportWorkerFailure [failed](https://tanzanite.cluster.engflow.com/invocations/default/66eef19a-69b0-406c-aa14-39903901cf98?testReportRun=1&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvc3FsL2ltcG9ydGVyOmltcG9ydGVyX3Rlc3Q=) on master @ [067e48d29b9093038f6fcf2074cd761ffdcd4fe2](https://github.com/cockroachdb/cockroach/commits/067e48d29b9093038f6fcf2074cd761ffdcd4fe2):


```
=== RUN   TestImportWorkerFailure
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestImportWorkerFailure1728561906
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    import_stmt_test.go:5380: IMPORT INTO t CSV DATA ('http://127.0.0.1:39717/0', 'http://127.0.0.1:39717/1', 'http://127.0.0.1:39717/2', 'http://127.0.0.1:39717/3', 'http://127.0.0.1:39717/4', 'http://127.0.0.1:39717/5', 'http://127.0.0.1:39717/6', 'http://127.0.0.1:39717/7', 'http://127.0.0.1:39717/8', 'http://127.0.0.1:39717/9', 'http://127.0.0.1:39717/10', 'http://127.0.0.1:39717/11', 'http://127.0.0.1:39717/12', 'http://127.0.0.1:39717/13', 'http://127.0.0.1:39717/14', 'http://127.0.0.1:39717/15', 'http://127.0.0.1:39717/16', 'http://127.0.0.1:39717/17', 'http://127.0.0.1:39717/18', 'http://127.0.0.1:39717/19') WITH sstsize = '1B' failed, checking that imported data was completely removed: "pq: job 952221520862117889: node liveness error: restarting in background"
    import_stmt_test.go:5381: error executing 'SELECT * FROM t ORDER BY i': pq: relation "t" is offline: importing
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestImportWorkerFailure1728561906
--- FAIL: TestImportWorkerFailure (217.30s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestImportWorkerFailure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-36771

sql/importer.TestImportWorkerFailure [failed](https://tanzanite.cluster.engflow.com/invocations/default/20f00515-0268-484c-aae2-50920fd99327?testReportRun=3&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvc3FsL2ltcG9ydGVyOmltcG9ydGVyX3Rlc3Q=) on master @ [2a5e231716c436781f12452d800651f51c6383b7](https://github.com/cockroachdb/cockroach/commits/2a5e231716c436781f12452d800651f51c6383b7):


```
=== RUN   TestImportWorkerFailure
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestImportWorkerFailure3401540108
    test_log_scope.go:81: use -show-logs to present logs inline
    jobs_verification.go:98: job failed: result is ambiguous: server shutdown
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestImportWorkerFailure3401540108
--- FAIL: TestImportWorkerFailure (109.79s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestImportWorkerFailure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql/importer.TestImportWorkerFailure [failed](https://tanzanite.cluster.engflow.com/invocations/default/3f5f12a7-5e12-438a-92ab-c56a7e833800?testReportRun=1&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvc3FsL2ltcG9ydGVyOmltcG9ydGVyX3Rlc3Q=) on master @ [4f78184a9e41aeaa24a2dfcb7462ab01dfa4703d](https://github.com/cockroachdb/cockroach/commits/4f78184a9e41aeaa24a2dfcb7462ab01dfa4703d):


```
=== RUN   TestImportWorkerFailure
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestImportWorkerFailure3079729921
    test_log_scope.go:81: use -show-logs to present logs inline
    jobs_verification.go:98: job failed: result is ambiguous: server shutdown
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestImportWorkerFailure3079729921
--- FAIL: TestImportWorkerFailure (113.79s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestImportWorkerFailure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/sql/importer/import_stmt_test.go
+++ b/pkg/sql/importer/import_stmt_test.go
@@ -5365,7 +5365,8 @@ func TestImportWorkerFailure(t *testing.T) {
 		t.Fatalf("%s: query returned before expected: %s", err, query)
 	}
 	var jobID jobspb.JobID
-	sqlDB.QueryRow(t, `SELECT id FROM system.jobs ORDER BY created DESC LIMIT 1`).Scan(&jobID)
+	sqlDB.QueryRow(t, `SELECT id FROM system.jobs WHERE status = 'running'
+                             AND job_type = 'IMPORT' ORDER BY created DESC LIMIT 1`).Scan(&jobID)
 
 	// Shut down a node.
 	tc.StopServer(1)
@@ -5378,13 +5379,13 @@ func TestImportWorkerFailure(t *testing.T) {
 	// rolled back.
 	if err := <-errCh; err != nil {
 		t.Logf("%s failed, checking that imported data was completely removed: %q", query, err)
-		sqlDB.CheckQueryResults(t, `SELECT * FROM t ORDER BY i`, [][]string{})
+		sqlDB.CheckQueryResultsRetry(t, `SELECT * FROM t ORDER BY i`, [][]string{})
 		return
 	}
 
 	// But the job should be restarted and succeed eventually.
 	jobutils.WaitForJobToSucceed(t, sqlDB, jobID)
-	sqlDB.CheckQueryResults(t,
+	sqlDB.CheckQueryResultsRetry(t,
 		`SELECT * FROM t ORDER BY i`,
 		sqlDB.QueryStr(t, `SELECT * FROM generate_series(0, $1)`, count-1),
 	)

